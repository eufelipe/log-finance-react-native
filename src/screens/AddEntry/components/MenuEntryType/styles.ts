import styled from 'styled-components/native';
import {TouchableOpacity, Text} from 'styles/layout';

export const Container = styled.View`
  flex-direction: row;
  padding-top: ${({theme}) => theme.spacing.small}px;
`;

export const Touchable = styled(TouchableOpacity)`
  margin: 0px ${({theme}) => theme.spacing.great}px
    ${({theme}) => theme.spacing.tall}px;
`;

export const LineActive = styled.View<{active: boolean}>`
  height: 3px;
  width: 30px;
  align-self: center;
  border-radius: 2px;
  margin: 8px 0px 10px;
  background: ${({theme, active = false}) =>
    active ? theme.colors.white : 'transparent'};
`;

export const Label = styled(Text)<{active: boolean}>`
  font-family: ${({theme}) => theme.fontFamily.regular};
  font-size: ${({theme}) => theme.fontSizes.medium};
  font-weight: ${({theme}) => theme.fontWeight.bold};
  color: ${({theme, active = false}) =>
    active ? theme.colors.white : theme.colors.gray};
`;
