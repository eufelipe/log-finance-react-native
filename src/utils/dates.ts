import {
  format as formatPattern,
  isValid,
  parseISO as parseISOFNS,
} from 'date-fns';

import locale from 'date-fns/locale/pt-BR';

export const DATE_FORMAT_ISO = 'yyyy-MM-dd';
export const DATE_FORMAT_BR = 'dd/MM/yyyy';
export const DATE_FORMAT_FRIENDLY = " dd/MM 'às' HH:mm";
export const DATE_FORMAT_FRIENDLY_TIME = "'às' HH:mm";

export function parseStringToDate(value: string): Date {
  const parsedDate = parseISOFNS(value);
  if (!isValid(parsedDate)) {
    throw Error('canot parse date');
  }
  return parsedDate;
}
export function parseDateToString({
  value,
  format = DATE_FORMAT_BR,
}: {
  value: Date;
  format?: string;
}): string {
  return formatPattern(value, format, {locale});
}

export const getDate = (date: Date): string => {
  return formatPattern(date, DATE_FORMAT_ISO, {locale});
};

export const getDateToday = (): string => {
  const today = new Date();
  return formatPattern(today, DATE_FORMAT_ISO, {locale});
};

export const getDateAndHourFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY, {locale});
};

export const getTimeFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY_TIME, {locale});
};
