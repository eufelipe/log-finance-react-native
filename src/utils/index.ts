export {default as number} from './number';
export {default as strings} from './strings';
