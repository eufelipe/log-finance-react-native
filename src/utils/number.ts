export const sanitizeToFloat = (value: string): number => {
  value = value.replace(/[^0-9.]+/, '');
  if (value.includes('.') && value.includes(',')) {
    value = value.replace(/\./g, '').replace(',', '.');
  } else if (value.includes(',')) {
    value = value.replace(',', '.');
  }
  return parseFloat(value) || 0;
};
