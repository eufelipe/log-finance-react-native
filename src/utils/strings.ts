export const sanitizeString = (value: string): string => {
  return value
    .toLowerCase()
    .replace(/á/g, 'a')
    .replace(/é/g, 'e')
    .replace(/í/g, 'i')
    .replace(/ó/g, 'o')
    .replace(/ú/g, 'u');
};
