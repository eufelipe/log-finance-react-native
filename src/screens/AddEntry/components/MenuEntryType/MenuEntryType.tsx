import React from 'react';

import {useTranslation} from 'react-i18next';
import {useEntry} from 'hooks/useEntry';
import {Container, Touchable, Label, LineActive} from './styles';

const MenuEntryType = (): JSX.Element => {
  const {entryType, setEntryType} = useEntry();

  const {t} = useTranslation('add');

  const isExpense = entryType === 'expense';
  const isEarning = entryType === 'earning';

  return (
    <Container>
      <Touchable onPress={() => setEntryType('expense')}>
        <Label active={isExpense}>{t('expense')}</Label>

        <LineActive active={isExpense} />
      </Touchable>
      <Touchable onPress={() => setEntryType('earning')}>
        <Label active={isEarning}>{t('earning')}</Label>
        <LineActive active={isEarning} />
      </Touchable>
    </Container>
  );
};

export default MenuEntryType;
