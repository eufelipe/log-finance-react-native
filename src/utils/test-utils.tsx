/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render, RenderAPI, RenderOptions} from '@testing-library/react-native';

import {theme} from 'styles/theme';
import {ThemeProvider} from 'styled-components/native';

import {NavigationContainer} from '@react-navigation/native';

interface Props {
  children: JSX.Element;
}

const AllTheProviders = ({children}: Props) => {
  return (
    <ThemeProvider theme={theme}>
      <NavigationContainer>{children}</NavigationContainer>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions,
): RenderAPI => render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
