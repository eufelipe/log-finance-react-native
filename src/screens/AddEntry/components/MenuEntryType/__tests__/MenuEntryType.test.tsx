/* eslint-disable @typescript-eslint/unbound-method */
/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render} from 'utils/test-utils';
import MenuEntryType from '../MenuEntryType';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({t: (key: any) => key}),
}));

describe('MenuEntryType', () => {
  it('Test match snapshot MenuEntryType', () => {
    // given
    const props = {};

    // when
    const {toJSON} = render(<MenuEntryType {...props} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });
});
