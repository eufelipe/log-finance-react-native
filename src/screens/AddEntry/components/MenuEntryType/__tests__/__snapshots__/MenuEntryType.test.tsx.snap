// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuEntryType Test match snapshot MenuEntryType 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "paddingTop": 10.714285714285714,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginBottom": 42.857142857142854,
        "marginLeft": 21.428571428571427,
        "marginRight": 21.428571428571427,
        "marginTop": 0,
        "opacity": 1,
      }
    }
  >
    <Text
      active={false}
      style={
        Array [
          Object {
            "color": "#464E5F",
            "fontFamily": "Poppins-Regular",
            "fontSize": 36,
            "fontWeight": "700",
          },
        ]
      }
    >
      expense
    </Text>
    <View
      active={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderRadius": 2,
            "height": 3,
            "marginBottom": 10,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 8,
            "width": 30,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginBottom": 42.857142857142854,
        "marginLeft": 21.428571428571427,
        "marginRight": 21.428571428571427,
        "marginTop": 0,
        "opacity": 1,
      }
    }
  >
    <Text
      active={false}
      style={
        Array [
          Object {
            "color": "#464E5F",
            "fontFamily": "Poppins-Regular",
            "fontSize": 36,
            "fontWeight": "700",
          },
        ]
      }
    >
      earning
    </Text>
    <View
      active={false}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderRadius": 2,
            "height": 3,
            "marginBottom": 10,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 8,
            "width": 30,
          },
        ]
      }
    />
  </View>
</View>
`;
